### PR TITLE
WT-7381 Cache btree's ckptlist between checkpoints

### PR DESCRIPTION
**Pull request by sulabhM**

### CODE REVIEW

**Comment by sulabhM** [comment]
@jiechenbo, I had some merge conflicts when I merged develop to this change. Please review the usage around `__meta_blk_mods_load`. Note I renamed it to match our convention of function names starting with `__`, and added static keyword where the function is defined.

**Review by jiechenbo — APPROVED** [review]
lgtm!

**Comment by sulabhM** [comment]
@keithbostic , can you have a look at the last changes I made to this PR.

I have run an extended set of WT tests, the ones that failed are known and unrelated to this change.

**Review by keithbostic — APPROVED** [review]
lgtm

File: src/meta/meta_ckpt.c [hunk 1]
```diff
@@ -503,42 +537,224 @@ meta_blk_mods_load(WT_SESSION_IMPL *session, const char *config, WT_CKPT *ckpt,
...
+    WT_ASSERT(session,
+      (ckpt_a->addr.data == NULL && ckpt_b->addr.data == NULL) ||
+        (ckpt_a->addr.data != NULL && ckpt_b->addr.data != NULL &&
+          memcmp(ckpt_a->addr.data, ckpt_b->addr.data,
+            WT_MIN(ckpt_a->addr.size, ckpt_b->addr.size)) == 0));
+    WT_ASSERT(session,
+      (ckpt_a->raw.data == NULL && ckpt_b->raw.data == NULL) ||
+        (ckpt_a->raw.data != NULL && ckpt_b->raw.data != NULL &&
+          memcmp(ckpt_a->raw.data, ckpt_b->raw.data, WT_MIN(ckpt_a->raw.size, ckpt_b->raw.size)) ==
+            0));
```
Comment (luke-pearson) [inline src/meta/meta_ckpt.c:597]: Is there a reason we don't check that .size fields match? Using `WT_MIN` here seems odd.

**Pull request merged**